api/v1beta1/openstackdeploy_webhook.go

```diff
@@ -52,3 +65,142 @@ func (r *OpenStackDeploy) Default() {
+	nncpList := &nmstatev1alpha1.NodeNetworkConfigurationPolicyList{}
+	listOpts := []client.ListOption{
+		client.MatchingLabels(
+			map[string]string{
+				OpenStackNetConfigReconcileLabel: osnetcfgName,
+			},
+		),
+	}
+
+	if err := webhookClient.List(context.TODO(), nncpList, listOpts...); err != nil {
+		return err
+	}
+
+	if len(nncpList.Items) == 0 {
+		return fmt.Errorf("no NodeNetworkConfigurationPolicy found for reference label %s %s",
+			OpenStackNetConfigReconcileLabel,
+			osnetcfgName)
+	}
+
+	nncpReadyCount := 0
+	for _, nncp := range nncpList.Items {
+		if nncp.Status.Conditions != nil && len(nncp.Status.Conditions) > 0 {
+			condition := nmstate.GetCurrentCondition(nncp.Status.Conditions)
+			if condition != nil {
+				if condition.Type == nmstateshared.NodeNetworkConfigurationPolicyConditionAvailable &&
+					condition.Reason == nmstateshared.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured {
+
+					openstackdeploylog.Info(fmt.Sprintf("%s %s: %s", nncp.Kind, nncp.Name, condition.Message))
+					nncpReadyCount++
+				} else if condition.Type == nmstateshared.NodeNetworkConfigurationPolicyConditionDegraded {
+
+					// get NNCE list to receive more details on why nncp is Degraded
+					nnceList := &nmstatev1alpha1.NodeNetworkConfigurationEnactmentList{}
+					listOpts := []client.ListOption{
+						client.MatchingLabels(
+							map[string]string{
+								nmstateshared.EnactmentPolicyLabel: nncp.Name,
+							},
+						),
+					}
+
+					if err := webhookClient.List(context.TODO(), nnceList, listOpts...); err != nil {
+						return err
+					}
+
+					var nnceError string
+					var nnce nmstatev1alpha1.NodeNetworkConfigurationEnactment
+					for _, nnce = range nnceList.Items {
+						condition := nmstate.GetCurrentCondition(nnce.Status.Conditions)
+
+						// if the first nnce has condition.Reason == nmstateshared.NodeNetworkConfigurationEnactmentConditionFailedToConfigure
+						// return the message as a hint to look at, do not return all nnce messages
+						if condition != nil && condition.Reason == nmstateshared.NodeNetworkConfigurationEnactmentConditionFailedToConfigure {
+							nnceError = condition.Message
+							break
+						}
+					}
+
+					msg := fmt.Sprintf("validation failed: %s %s not in %s state: %s %s - NodeNetworkConfigurationEnactments %s message: %s",
+						nncp.GroupVersionKind().Kind,
+						nncp.GetName(),
+						nmstateshared.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured,
+						condition.Reason,
+						condition.Message,
+						nnce.Name,
+						nnceError,
+					)
+
+					// if user skip validation, just log the failed validation
+					if r.Spec.SkipNNCPValidation {
+						openstackdeploylog.Info(fmt.Sprintf("SkipNNCPValidation configured! %s", msg))
+					} else {
+						info := "It's recommended to resolve the failures before proceeding with the deployment. If this validation should be skipped add spec.SkipNNCPValidationvalidate: true"
```

Comment:
s/SkipNNCPValidationvalidate/SkipNNCPValidation

Reply:
done, also added it to the check to ValidateUpdate